test/functional/legacy/fnamemodify_spec.lua

```diff
@@ -16,17 +14,27 @@ describe('filename modifiers', function()
 
     source([=[
       func Test_fnamemodify()
-        let tmpdir = resolve('/tmp')
+        if has('win32')
+          let tmpdir = substitute($TMP, '\\', '/', 'g')
+          set shellslash
+        else
+          let tmpdir = resolve('/tmp')
+          set shell=sh
+        endif
         execute 'cd '. tmpdir
-        set shell=sh
-        set shellslash
         let $HOME=fnamemodify('.', ':p:h:h:h')
         call assert_equal('/', fnamemodify('.', ':p')[-1:])
-        call assert_equal('p', fnamemodify('.', ':p:h')[-1:])
+        if has('win32')
+          call assert_equal('p', fnamemodify('.', ':p:h:h')[-1:])
+        else
+          call assert_equal('p', fnamemodify('.', ':p:h')[-1:])
+        endif
         call assert_equal('t', fnamemodify('test.out', ':p')[-1:])
         call assert_equal('test.out', fnamemodify('test.out', ':.'))
         call assert_equal('../testdir/a', fnamemodify('../testdir/a', ':.'))
-        call assert_equal('test.out', fnamemodify('test.out', ':~'))
+        if !has('win32')
+          call assert_equal('test.out', fnamemodify('test.out', ':~'))
```

Comment:
This should work on Windows. Need at least a TODO comment.

Reply:
I can reproduce in Vim 8.0.1123.

Reply:
Ok, but I don't see why it shouldn't work on Windows ... can be an enhancement later.

Reply:
`%TMP%` uses dos path.
Must use `system` to convert `%TMP%` to the full path or try out `%$LOCALAPPDATA%\Temp`